File: logical test.c
# include <stdio.h>
# include <string.h>
int main()
{
    char str[100];
    int i, len, string = 1;
    printf("Enter a string: ");
    scanf("%s", str);
    len = strlen(str);
    for (i = 0; i < len / 2; i++) 
    {
        if (str[i] != str[len - i - 1]) {
            string= 0;
            break;
        }
    }

    if (string)
        printf("Palindrome\n");
    else
        printf("Not Palindrome\n");

    return 0;
}

File: logical test.py
n = 5
for i in range(n):
    print(' ' * (n - i - 1) + '*' * (2 * i + 1))
for i in range(n - 2, -1, -1):
    print(' ' * (n - i - 1) + '*' * (2 * i + 1))







def palindrome(s):
    i = 0
    j = len(s) - 1
    while i < j:
        if s[i] != s[j]:
            return False
        i += 1
        j -= 1
    return True
s = input("Enter a string: ")
if palindrome(s):
    print("Palindrome")
else:
    print("Not Palindrome")




